wolfcrypt/src/asn.c

```diff
@@ -14394,6 +14359,19 @@ static int ConfirmNameConstraints(Signer* signer, DecodedCert* cert)
             case ASN_RFC822_TYPE:
                 /* Shouldn't it validade E= in subject as well? */
                 name = cert->altEmailNames;
+                if (cert->subjectEmail != NULL) { /* add subject email to list*/
```

Comment:
* Can you comment on what happens when the `cert->altEmailNames` hasn't been allocated? Would like there to be a better inline code comment here.
* The `subjectDnsName` variable is on the stack... it can't be added to the list like this can it? Should it be allocated?
* How are you making sure `subjectDnsName` in the list is not trying to be free'd?